src/network/core/packet.h

```diff
@@ -83,6 +85,14 @@ struct Packet {
 	uint32 Recv_uint32();
 	uint64 Recv_uint64();
 	void   Recv_string(char *buffer, size_t size, StringValidationSettings settings = SVS_REPLACE_WITH_QUESTION_MARK);
+
+	/* Transfering data in to or out of this packet to some other function.
+	 * Think about TCP's recv and send, or UDP's recvfrom and sendto.
+	 * These are special as even though we can send or receive an amount of
+	 * data, those functions can say they only processed a smaller amount,
+	 * so special handling is required to keep the position pointers right. */
```

Comment:
The main problem is that it holds for all the transfer related function ;( So I'm not quite sure where it would fit best; maybe the RemainingBytesToTransfer since that is used in either case? I'm not really sure.

Reply:
Maybe in `TransferIn` and a comment in `TransferOut` stating to look at the doc of `TransferIn` ?